src/Generator.php

```diff
@@ -149,12 +158,45 @@ protected function scanFilesForDocumentation(): self
     {
         $this->openApi = openApiScan(
             $this->annotationsDir,
-            ['exclude' => $this->excludedDirs]
+            $this->getScanOptions()
         );
 
         return $this;
     }
 
+    /**
+     * Prepares options array for scanning files.
+     *
+     * @return array
+     */
+    protected function getScanOptions(): array
+    {
+        $options['exclude'] = $this->excludedDirs;
+
+        $paterns = Arr::get($this->scanOptions, 'pattern', []);
+        if (! empty($paterns)) {
+            $options['pattern'] = $paterns;
+        }
+
+        $processorClasses = Arr::get($this->scanOptions, 'processors', []);
+        $processors = [];
+
+        foreach (\OpenApi\Analysis::processors() as $processor) {
+            $processors[] = $processor;
+            if ($processor instanceof \OpenApi\Processors\BuildPaths) {
+                foreach ($processorClasses as $customProcessor) {
+                    $processors[] = new $customProcessor();
+                }
+            }
+        }
```

Comment:
Can you provide an example in config how to declare these custom processors?

Reply:
Here is the example I'm starting from:
https://github.com/zircote/swagger-php/tree/master/Examples/schema-query-parameter-processor

I mentioned it in the issue

And my processor is half-ready (I add model properties automatically).
It's in the middle of the work, so too much garbage, but you can get the idea if official processor example is not enough for you :-)

```
<?php

namespace Infrastructure\SwaggerProcessors;

use OpenApi\Analysis;
use Illuminate\Support\Arr;
use Illuminate\Support\Str;
use const OpenApi\UNDEFINED;
use OpenApi\Annotations\Schema;
use OpenApi\Annotations\Property;
use OpenApi\Annotations\Response;
use OpenApi\Annotations\MediaType;
use OpenApi\Annotations\Operation;
use OpenApi\Annotations\Parameter;
use OpenApi\Annotations\Components;
use OpenApi\Annotations\JsonContent;
use OpenApi\Annotations\AbstractAnnotation;

/**
 * Custom processor to translate the vendor tag `query-args-$ref` into query parameter annotations.
 *
 * Details for the parameters are taken from the referenced schema.
 */
class SchemaQueryParameter
{
    const X_QUERY_AGS_REF = 'query-args-$ref';
    const MODEL_INPUT_FIELDS = 'model-input-fields';
    const PROPERY_FINAL = 'final';

    public function __invoke(Analysis $analysis)
    {

        $schemas = $analysis->getAnnotationsOfType(Schema::class, true);

        // $mediaTypes = $analysis->getAnnotationsOfType(MediaType::class);
        $mediaTypes = $analysis->getAnnotationsOfType(JsonContent::class);
        // dd($mediaType);
        foreach ($mediaTypes as $mediaType) {
            if ($mediaType->x !== UNDEFINED && array_key_exists(self::MODEL_INPUT_FIELDS, $mediaType->x)) {
                foreach ($mediaType->x[self::MODEL_INPUT_FIELDS] as $ref) {
                    if ($schema = $this->schemaForRef($schemas, $ref)) {
                        $this->expandAnnotation($mediaType, $schema);
                        // $this->expandQueryArgs($mediaType, $schema);
                        $this->cleanUp($mediaType);
                    }
                }
            }
        }
        return;

        $operations = $analysis->getAnnotationsOfType(Operation::class);
        foreach ($operations as $operation) {
            if ($operation->x !== UNDEFINED && array_key_exists(self::X_QUERY_AGS_REF, $operation->x)) {
                dd('bbdb', $operation->x);
                if ($schema = $this->schemaForRef($schemas, $operation->x[self::X_QUERY_AGS_REF])) {
                    $this->expandQueryArgs($operation, $schema);
                    $this->cleanUp($operation);
                }
            }
        }
    }

    /**
     * Find schema for the given ref.
     */
    protected function schemaForRef(array $schemas, string $ref)
    {
        foreach ($schemas as $schema) {
            if (Components::SCHEMA_REF . $schema->schema === $ref) {
                return $schema;
            }
        }

        return null;
    }

    /**
     * Expand the given operation by injecting parameters for all properties of the given schema.
     */
    protected function expandAnnotation(AbstractAnnotation $annotation, Schema $schema)
    {
        if ($schema->properties === UNDEFINED || !$schema->properties) {
            return;
        }
        if ('Domain' !== $schema->schema) {
            return;
        }

        $annotation->properties = $annotation->properties === UNDEFINED ? [] : $annotation->properties;

        $modelClassName = $schema->_context->getRootContext()->namespace . '\\' . $schema->_context->class;

        /**
         * @var \Infrastructure\Database\Eloquent\Model
         */
        $model = new $modelClassName;


        // How to get indexes
        // $tableName = $model->getTable();
        // $indexColumns = \DB::getDoctrineSchemaManager()->listTableIndexes($tableName);
        // foreach ($indexColumns as $index) {
        //     $index->isUnique();
        //     $index->isPrimary();
        // }
        // d($indexColumns);
//         $collection = collect($schema->properties);
//         $a = $collection->where('property', 'domain_parent_uuid')->first();

// dd($a, $a->x);
        $columns = $model->getTableColumnsInfo(true);
        foreach ($columns as $columnName => $column) {
            if (!$this->processPropery($schema, $model, $columnName, false)) {
                continue;
            }

            $options = $this->mapDBTypesToOATypes($column->getType()->getName());
            $options['property'] = $columnName . uniqid();
            $newPropery = new Property($options);
            $annotation->properties[] = $newPropery;
        }

        return;


        // $annotation->responses = $annotation->responses === UNDEFINED ? [] : $annotation->responses;
        $annotation->properties = $annotation->properties === UNDEFINED ? [] : $annotation->properties;
        // dd($schema);
        $i = 200;
        foreach ($schema->properties as $property) {
            dd($property);
            $newPropery = new Property([
                'property' => 'solme',
                'schema' => 'integer',
                'example' => 3,
            ]);
            $annotation->properties[] = $newPropery;
            return;
            continue;

            $response = new Response([
                'response' => ++$i,
                'description' => 'aaa',
                // 'content' => $property->property,
                // 'name' => $property->property,
                // 'in' => 'query',
                // 'required' => false,
                // 'schema' => [
                //     'type' => 'integer',
                //     'format' => 'int64',
                // ]
            ]);
            $annotation->responses[] = $response;
        }

        return;

        $annotation->parameters = $annotation->parameters === UNDEFINED ? [] : $annotation->parameters;

        foreach ($schema->properties as $property) {
            $parameter = new Parameter([
                'name' => $property->property,
                'in' => 'query',
                'required' => false,
                'schema' => [
                    'type' => 'integer',
                    'format' => 'int64',
                ]
            ]);
            $annotation->parameters[] = $parameter;
        }
    }

    /**
     * Expand the given operation by injecting parameters for all properties of the given schema.
     */
    protected function expandQueryArgs(Operation $operation, Schema $schema)
    {
        if ($schema->properties === UNDEFINED || !$schema->properties) {
            return;
        }
        if ('Domain' !== $schema->schema) {
            return;
        }
        return;

        // dd(count($operation->responses));
        // dd($operation->responses);
        $operation->responses = $operation->responses === UNDEFINED ? [] : $operation->responses;
        // dd($schema);
        $i = 200;
        foreach ($schema->properties as $property) {
            $response = new Response([
                'response' => ++$i,
                'description' => 'aaa',
                // 'content' => $property->property,
                // 'name' => $property->property,
                // 'in' => 'query',
                // 'required' => false,
                // 'schema' => [
                //     'type' => 'integer',
                //     'format' => 'int64',
                // ]
            ]);
            $operation->responses[] = $response;
        }

        return;

        $operation->parameters = $operation->parameters === UNDEFINED ? [] : $operation->parameters;

        foreach ($schema->properties as $property) {
            $parameter = new Parameter([
                'name' => $property->property,
                'in' => 'query',
                'required' => false,
                'schema' => [
                    'type' => 'integer',
                    'format' => 'int64',
                ]
            ]);
            $operation->parameters[] = $parameter;
        }
    }

    /**
     * Clean up.
     */
    protected function cleanUp($operation)
    {
        unset($operation->x[self::X_QUERY_AGS_REF]);
        if (!$operation->x) {
            $operation->x = UNDEFINED;
        }
    }

    private function mapDBTypesToOATypes($type) {
        $mapping = [
            "decimal" => [
                'type' => 'number',
            ],
            "text" => [
                'type' => 'string',
            ],
            "guid" => [
                'type' => 'string' ,
                'format' => 'uuid' ,
                'example' => Str::uuid()->toString(),
            ],
        ];

        $mappedType = Arr::get($mapping, $type, []);
        if (empty($mappedType)) {
            $mappedType['type'] = $type;
        }

        return $mappedType;
    }

    private function processPropery($schema, $model, $columnName, $fillabeOnly = true) {
        if ($fillabeOnly) {
            $fillable = $model->getFillable();
            if (!in_array($columnName, $fillable)) {
                return false;
            }
        }

        $collection = collect($schema->properties);
        $propety = $collection->where('property', $columnName)->first();
        if (!$propety) {
            return true;
        }

        $propety->x = $propety->x === UNDEFINED ? [] : $propety->x;

        $isFinal = Arr::get($propety->x, self::PROPERY_FINAL, false);

        if ($isFinal === true) {
            return false;
        }

        return true;
    }
}
```

And comment below, currently the processor hooks is this place

```
                    x={
                        "model-input-fields"={
                            "#/components/schemas/Domain",
                            "#/components/schemas/User",
                        },
                    },
```

```
   /**
    @OA\Post(
        tags={"Domain", "User"},
        x={"query-args-$ref"="#/components/schemas/Domain"},
        security={"api_key"},
        path="/signup/domain",
        summary="Create a domain",
        description="Creates a domain. Depending on the configuration the domain `domain_name` can be a subdomain of the main domain or an independent domain",
        @OA\RequestBody(
            description="Client side search object",
            required=true,
            @OA\JsonContent(
                allOf={
                    @OA\Schema(ref="#/components/schemas/Domain"),
                    @OA\Schema(
                        @OA\Property(property="iidd", type="integer"),
                        @OA\Property(property="created_at", ref="#/components/schemas/Domain/properties/domain_uuid")
                        )
                    },
                    x={
                        "model-input-fields"={
                            "#/components/schemas/Domain",
                            "#/components/schemas/User",
                        },
                    },
            ),
        ),
        @OA\Response(
        response=200,
        description="Get application name and version",
        @OA\JsonContent(
                allOf={
                    @OA\Schema(ref="#/components/schemas/Domain"),
                    @OA\Schema(
                        @OA\Property(property="iidd", type="integer"),
                        @OA\Property(property="created_at", ref="#/components/schemas/Domain/properties/domain_uuid")
                    )
                }
            ),
        ),
    )
     */

Reply:
```
            'processors' => [
                \Infrastructure\SwaggerProcessors\SchemaQueryParameter::class,
            ],
```

Reply:
I'm not sure what you are asking about. An example is the comment in the config file. Please, clarify what you mean.

Reply:
I just don't understand. You loop through `\OpenApi\Analysis::processors()` and adding those to array. And if processor from this list is `instanceof \OpenApi\Processors\BuildPaths` you starting to add processors from the configuration. Why? If we need to keep/pass existing processors can we just do `array_merge`? Or I am missing something here?

Reply:
I took this part from official example
https://github.com/zircote/swagger-php/blob/master/Examples/schema-query-parameter-processor/scan.php#L9

I suspect they have reasons for such an approach. I didn't investigate why.  I suspect there are different types of processors and we must add only if  \OpenApi\Processors\BuildPaths is among current processors.

Reply:
No. It seems, that we inject our processors after instanceof \OpenApi\Processors\BuildPaths and before anything else.
It rebuilds processor list with our custom processors inside it, somewhere in the middle.